### PR TITLE
Peertube: add aarch64-linux support

### PR DESCRIPTION
**Pull request by cillianderoiste**
###### Description of changes

This adds support for aarch64-linux to peertube. I've tested and am running a backported version of this on an RPi 4 (based on 22.05). 
I removed the if statement which threw an error if you attempted to install peertube on an unsupported platform since I suspect this is already covered in the meta.platforms list, but please correct me if I'm wrong.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Izorkin** [comment]
Required changes:
```diff
diff --git a/nixos/tests/all-tests.nix b/nixos/tests/all-tests.nix
index e577001a3ba..964427a46ba 100644
--- a/nixos/tests/all-tests.nix
+++ b/nixos/tests/all-tests.nix
@@ -499,7 +499,7 @@ in {
   pdns-recursor = handleTest ./pdns-recursor.nix {};
   peerflix = handleTest ./peerflix.nix {};
   peering-manager = handleTest ./web-apps/peering-manager.nix {};
-  peertube = handleTestOn ["x86_64-linux"] ./web-apps/peertube.nix {};
+  peertube = handleTestOn ["aarch64-linux" "x86_64-linux"] ./web-apps/peertube.nix {};
   pgadmin4 = handleTest ./pgadmin4.nix {};
   pgadmin4-standalone = handleTest ./pgadmin4-standalone.nix {};
   pgjwt = handleTest ./pgjwt.nix {};

```

**Comment by Izorkin** [comment]
@GrahamcOfBorg test peertube

**Review by Izorkin — APPROVED** [review]

File: pkgs/servers/peertube/default.nix [hunk 1]
```diff
@@ -2,14 +2,21 @@
...
+  bcrypt_arch_details = {
+    "x86_64-linux" = {
+      "arch" = "linux-x64";
+      "hash" = "sha256-I1ceMi7h6flvKBmMIU1qjAU1S6z5MzguHDul3g1zMKw=";
+    };
+    "aarch64-linux" = {
+      arch = "linux-arm64";
+      hash = "sha256-q8BR7kILYV8i8ozDkpcuKarf4s1TgRqOrUeLqjdWEQ0=";
+    };
+  };
```
Comment (SuperSandro2000) [inline pkgs/servers/peertube/default.nix:14]: Missing or unsupported system

Comment (cillianderoiste) [reply pkgs/servers/peertube/default.nix:14]: I don't understand, do you expect arch = "aarch64-linux"? The string for "arch" is only used as part of the URL for bcrypt_lib below.

**Comment by cillianderoiste** [comment]
It would be great to get this merged. Is there anything I can do to help push it along?

File: pkgs/servers/peertube/default.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/servers/peertube/default.nix:14]: This code is not evaling on any other platform which it should. see eg https://github.com/NixOS/nixpkgs/blob/68196a61c26748d3e53a6803de3d2f8c69f27831/pkgs/applications/networking/browsers/tor-browser-bundle-bin/default.nix#L120

**Pull request closed without merging**